contracts/utils/math/SafeCast.sol

```diff
@@ -17,6 +17,326 @@ pragma solidity ^0.8.19;
  * class of bugs, so it's recommended to use it always.
  */
 library SafeCast {
```

Comment:
I don't like the idea because `uint8 bitSize` allow for errors without invalid `bitSize` numbers.

Actually, these errors are currently my best candidates for `StdError`s

Reply:
I'm not really concerned about invalid bitSize values to be honest, I would prefer less code, smaller surface.

Reply:
Open to not having `bitSize` if that is a concern but still prefer a single error.

Reply:
Solved with the following errors:

```solidity
/**
 * @dev Value doesn't fit in an uint of `bits`.
 *
 * NOTE: The `bits` argument is `min(bits, 255)`.
 */
error SafeCastOverflowedUintDowncast(uint8 bits, uint256 value);

/**
 * @dev An int value doesn't fit in an uint of `bits`.
 *
 * NOTE: The `bits` argument is `min(bits, 255)`.
 */
error SafeCastOverflowedIntToUint(uint8 bits, int256 value);

/**
 * @dev Value doesn't fit in an int of `bits`.
 *
 * NOTE: The `bits` argument is `min(bits, 255)`.
 */
error SafeCastOverflowedIntDowncast(uint8 bits, int256 value);

/**
 * @dev An uint value doesn't fit in an int of `bits`.
 *
 * NOTE: The `bits` argument is `min(bits, 255)`.
 */
error SafeCastOverflowedUintToInt(uint8 bits, uint256 value);
```